writer.go

```diff
@@ -762,9 +762,23 @@ func (c *writerColumn) flushFilterPages() error {
 		if c.page.filter == nil {
 			c.page.filter = c.newBloomFilterEncoder(numValues)
 		}
-		for _, page := range c.filter {
-			if err := page.WriteTo(c.page.filter); err != nil {
-				return err
+
+		// This early return makes it safe to check the first page for the
+		// dictionary below.
+		if len(c.filter) == 0 {
+			return nil
+		}
+
+		// It's safe to assume that if the first page has a dictionary all will.
+		if c.filter[0].Dictionary() == nil {
+			for _, page := range c.filter {
+				if err := page.WriteTo(c.page.filter); err != nil {
+					return err
+				}
+			}
+		} else {
+			for _, page := range c.filter {
+				page.Dictionary().WriteTo(c.page.filter)
```

Comment:
How about this shorter form, would it do the same?
```suggestion
		// It's safe to assume that if the first page has a dictionary all will.
		if dict := c.dictionary; dict == nil {
			for _, page := range c.filter {
				if err := page.WriteTo(c.page.filter); err != nil {
					return err
				}
			}
		} else {
			for _, page := range c.filter {
				dict.WriteTo(c.page.filter)
```
We should probably check the error returned by `WriteTo` as well in the second loop.

Reply:
🤔 

Looking more at the code, do we need a loop in the second case? The dictionary will be the same for all pages, maybe we can only write it once to the filter, next calls to `WriteTo` would be no-ops.

Reply:
yeah I realized the same, pushed the change!